hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaSoftwareModuleManagement.java

```diff
@@ -358,25 +377,33 @@ public Slice<AssignedSoftwareModule> findAllOrderBySetAssignmentAndModuleNameAsc
 
         final List<AssignedSoftwareModule> resultList = new ArrayList<>();
 
-        smWithAssignedFlagList.forEach(smWithAssignedFlag -> resultList
-                .add(new AssignedSoftwareModule(smWithAssignedFlag.get("sm", JpaSoftwareModule.class),
-                        smWithAssignedFlag.get("assigned", Number.class).longValue() == 1)));
+        smWithAssignedFlagList.forEach(smWithAssignedFlag -> {
+            final JpaSoftwareModule softwareModule = smWithAssignedFlag.get("sm", JpaSoftwareModule.class);
+            try {
+                softwareModuleRepository.findById(softwareModule.getId()); // check read access
+                resultList
+                        .add(new AssignedSoftwareModule(softwareModule,
+                                smWithAssignedFlag.get("assigned", Number.class).longValue() == 1));
+            } catch (final InsufficientPermissionException e) {
+                // skip the entry
+            }
```

Comment:
What about perform an exists query using a specification by ID like:

```suggestion
            // check accessibility
            final JpaSoftwareModule softwareModule = smWithAssignedFlag.get("sm", JpaSoftwareModule.class);
            if (softwareModuleRepository.exists(SoftwareModuleSpecification.byId(softwareModule.getId()))) {
                resultList.add(new AssignedSoftwareModule(softwareModule,
                        smWithAssignedFlag.get("assigned", Number.class).longValue() == 1));
            }
```

Reply:
AFAIU the you point is skip - not to throw exception if any inaccessible. You are right - it should be like so. I change it to (skipping, possibly, db call):
```java
// check read access
if (softwareModuleRepository.getAccessController().map(accessController -> {
    try {
        accessController.assertOperationAllowed(AccessController.Operation.READ, softwareModule);
        return true;
    } catch (final InsufficientPermissionException e) {
        return false;
    }
}).orElse(true)) {
    resultList.add(new AssignedSoftwareModule(softwareModule,
            smWithAssignedFlag.get("assigned", Number.class).longValue() == 1));
}
```